api/types/infra.go

```diff
@@ -1,5 +1,10 @@
 package types
 
+// InfraSetupRequest represents a request to set up infrastructure
+type InfraSetupRequest struct {
+	IgnorePreflightFailures bool `json:"ignorePreflightFailures"`
```

Comment:
the cli flag and variable name we pass to the api is named `allowIgnoreHostPreflights`, i think we should keep naming consistent and name this `ignoreHostPreflights`.